blockcache/blockcache.go

```diff
@@ -0,0 +1,59 @@
+package blockcache
+
+import (
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
+	"github.com/lightninglabs/neutrino/cache"
+	"github.com/lightninglabs/neutrino/cache/lru"
+)
+
+type BlockCache struct {
+	Cache *lru.Cache
+}
+
+func NewBlockCache(capactity uint64) *BlockCache {
+	return &BlockCache{
+		Cache: lru.NewCache(capactity),
+	}
+}
+
+func (bc *BlockCache) GetBlock(hash *chainhash.Hash,
+	getBlockImpl func(hash *chainhash.Hash) (*wire.MsgBlock,
+		error)) (*wire.MsgBlock, error) {
+
+	// Create an inv vector for getting the block.
+	inv := wire.NewInvVect(wire.InvTypeWitnessBlock, hash)
```

Comment:
not sure if it is fine to hard code this...

Reply:
Should be fine, do we hard code it elsewhere in the codebase?